pkg/model/combine_merging.go

```diff
@@ -0,0 +1,152 @@
+package model
+
+import (
+	"bytes"
+	"hash/fnv"
+
+	"github.com/pkg/errors"
+
+	"github.com/grafana/tempo/pkg/tempopb"
+	"github.com/grafana/tempo/tempodb/encoding/common"
+)
+
+type ObjectMerger interface {
+	ID() common.ID
+	Merge(dataEncoding string, objs ...[]byte) (int, error)
+	Result(dataEncoding string) ([]byte, error)
+}
+
+type objectMerger struct {
```

Comment:
Is this a standard approach in this project, providing an exported interface and a private implementation, or are we doing this to allow the `instrumentedObjectMerger` wrapping?

We probably need some `combine_merging_test.go` tests.